### PR TITLE
[Fix](Status) Handle status code correctly and add a new error code `ENTRY_NOT_FOUND`

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes
coredump:
```
0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/common/signal_handler.h:413
1# 0x00007F7902A1C090 in /lib/x86_64-linux-gnu/libc.so.6
2# raise at ../sysdeps/unix/sysv/linux/raise.c:51
3# abort at /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:81
4# 0x0000556C9A5AEC69 in /mnt/hdd01/ci/branch20-deploy/be/lib/doris_be
5# 0x0000556C9A5A427D in /mnt/hdd01/ci/branch20-deploy/be/lib/doris_be
6# google::LogMessage::SendToLog() in /mnt/hdd01/ci/branch20-deploy/be/lib/doris_be
7# google::LogMessage::Flush() in /mnt/hdd01/ci/branch20-deploy/be/lib/doris_be
8# google::LogMessageFatal::~LogMessageFatal() in /mnt/hdd01/ci/branch20-deploy/be/lib/doris_be
9# doris::Tablet::calc_segment_delete_bitmap(std::shared_ptr, std::shared_ptr const&, std::vector<std::shared_ptr, std::allocator<std::shared_ptr > > const&, std::shared_ptr, long, doris::RowsetWriter*) in /mnt/hdd01/ci/branch20-deploy/be/lib/doris_be
10# doris::Tablet::calc_delete_bitmap(std::shared_ptr, std::vector<std::shared_ptr, std::allocator<std::shared_ptr > > const&, std::vector<std::shared_ptr, std::allocator<std::shared_ptr > > const&, std::shared_ptr, long, doris::CalcDeleteBitmapToken*, doris::RowsetWriter*) at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/tablet.cpp:3081
11# doris::MemTable::_generate_delete_bitmap(int) at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/memtable.cpp:480
12# doris::MemTable::_do_flush()::$_0::operator()(unsigned long) const at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/memtable.cpp:535
13# doris::Status std::_invoke_impl<doris::status, doris::memtable::_do_flush()::$_0&,="" int="">(std::_invoke_other, doris::MemTable::_do_flush()::$_0&, int&&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61
14# std::enable_if<is_invocable_r_v<doris::status, doris::memtable::do_flush()::$_0&,="" int="">, doris::Status>::type std::_invoke_r<doris::status, doris::memtable::_do_flush()::$_0&,="" int="">(doris::MemTable::_do_flush()::$_0&, int&&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:114
15# std::_Function_handler::_M_invoke(std::_Any_data const&, int&&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
16# std::function::operator()(int) const at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:560
17# doris::BetaRowsetWriter::flush_single_memtable(doris::vectorized::Block const*, long*, doris::FlushContext const*) at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/rowset/beta_rowset_writer.cpp:481
18# doris::MemTable::_do_flush() at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/memtable.cpp:540
19# doris::MemTable::flush() at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/memtable.cpp:503
20# doris::FlushToken::_flush_memtable(doris::MemTable*, long) at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/memtable_flush_executor.cpp:98
21# doris::MemtableFlushTask::run() at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/olap/memtable_flush_executor.cpp:46
22# doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/util/threadpool.cpp:531
23# void std::_invoke_impl<void, void="" (doris::threadpool::&)(),="" doris::threadpool&="">(std::_invoke_memfun_deref, void (doris::ThreadPool::&)(), doris::ThreadPool&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:74
24# std::_invoke_result::type std::_invoke(void (doris::ThreadPool::&)(), doris::ThreadPool&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96
25# void std::Bind::_call<void, ,="" 0ul="">(std::tuple<>&&, std::_Index_tuple<0ul>) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/functional:420
26# void std::_Bind::operator()<, void>() at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/functional:503
27# void std::_invoke_impl<void, std::_bind<void="" (doris::threadpool::(doris::threadpool))()="">&>(std::_invoke_other, std::_Bind&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61
28# std::enable_if<is_invocable_r_v<void, std::bind<void="" (doris::threadpool::(doris::threadpool))()="">&>, void>::type std::_invoke_r<void, std::_bind<void="" (doris::threadpool::(doris::threadpool))()="">&>(std::_Bind&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:117
29# std::_Function_handler >::_M_invoke(std::_Any_data const&) at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
30# std::function::operator()() const at /var/local/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:560
31# doris::Thread::supervise_thread(void*) at /home/zcp/repo_center/doris_branch-2.0/doris/be/src/util/thread.cpp:465
32# start_thread at /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:478
33# __clone at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
```

This PR handle the returned status code correctly after the modification of https://github.com/apache/doris/pull/23619 and add a more fine-grained new error code `ENTRY_NOT_FOUND` 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: be/src/common/status.h [hunk 1]
```diff
@@ -268,8 +268,9 @@ E(INVERTED_INDEX_NO_TERMS, -6005);
 E(INVERTED_INDEX_RENAME_FILE_FAILED, -6006);
 E(INVERTED_INDEX_EVALUATE_SKIPPED, -6007);
 E(INVERTED_INDEX_BUILD_WAITTING, -6008);
-E(KEY_NOT_FOUND, -6009);
-E(KEY_ALREADY_EXISTS, -6010);
```
Comment (dataroaring) [inline be/src/common/status.h:272]: DO NOT CHANGE existed code.

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**